### ISSUE TITLE
Woof and React Error

### ISSUE THREAD

**Issue by AlABarazi** [stevekrouse/WoofJS#519]
I add to head tag 
`<script src="https://cdn.rawgit.com/stevekrouse/WoofJS/b41555bf/dist/woof.js" global="false"></script>`
this should stop woof from making its method global.

I added to my code:
```
var project = new Woof({global: false, width: 300, height: 400})
var IMAGE_NAME = new project.Image({project: project, url: "./images/SMJjVCL.png"})
```

I got :

> Error: You must turn off global mode in the Woof script tag if you want to create your own Woof object.
> Woof — woof.js:206:138
> componentDidMount — ThirdPartyWoofjs.js:24
> commitLifeCycles — modules.js:23262
> commitAllLifeCycles — modules.js:24438
> callCallback — modules.js:15034
> dispatchEvent
> invokeGuardedCallbackDev — modules.js:15073
> invokeGuardedCallback — modules.js:14930
> commitRoot — modules.js:24542
> performWorkOnRoot — modules.js:25509
> performWork — modules.js:25459
> requestWork — modules.js:25370
> scheduleWorkImpl — modules.js:25224
> scheduleTopLevelUpdate — modules.js:25685
> updateContainer — modules.js:25723
> (anonymous function) — modules.js:29718
> renderSubtreeIntoContainer — modules.js:29717
> (anonymous function) — main.js:179
> maybeReady — meteor.js:809
> loadingCompleted — meteor.js:821
> logCapturedError — modules.js:24239

After removing global="false" from script tag these methods worked:
```
// setBackdropColor("blue")
var textSprite1 = new Text({
  text: () => "Hello world!",
  size: 12,
  color: "rgb(100, 50, 240)",
  fontFamily: "arial",
  textAlign: "left"
})



var circleSprite1 = new Circle({
  radius: 10,
  color: "red",
  x: 0,
  y: 0
})
var circleSprite1 = new Oval({
  radius: 10,
  color: "red",
  x: 0,
  y: 0
})
```

**Comment by stevekrouse** [comment]
Hi @alabrazi - sorry to hear you had trouble, but thanks for reaching out! I tried to reproduce the bug you described [here](https://jsfiddle.net/084b41wL/) but was unable to reproduce it - the code seemed to run without error. (The only thing I changed was adding `http://woofjs.com` to the image URL.)

Hope that solves your issue. If it does not, feel free to re-open this issue with a link to a page that demonstrates the bug.

**Comment by alabrazi** [comment]
Hi @stevekrouse  - Thank you for your comment! I checked the link that you have provided. It seems that in my case, this issue is maybe related to using Meteor, React and Node.js.

I have added to the main.html head tag the correct woofs script to main.html
![screen shot 2018-02-12 at 17 37 54](https://user-images.githubusercontent.com/7549404/36105057-568f9c44-101c-11e8-9344-6b064c031040.jpg)

Then made third party react component  with woofjs and tried another method to insert the script to the dom (in addition to adding it to main.html): 
![screen shot 2018-02-12 at 17 59 58](https://user-images.githubusercontent.com/7549404/36106000-c5872b6a-101e-11e8-9e6f-d15e1fec5f6e.jpg)

Both ways produced errors.
![screen shot 2018-02-12 at 17 41 05](https://user-images.githubusercontent.com/7549404/36105313-f6c19884-101c-11e8-81f6-09678ba1bff3.jpg)

I did the same process for [Pixijs](https://github.com/pixijs/pixi.js) and [Playcanvas engine](https://github.com/playcanvas/engine)  and both of them worked with Meteor as React components.

**Comment by stevekrouse** [comment]
Ah fascinating! I've never seen Woof used as part of a React project so this is a new issue. I'll re-open the issue with a new name...

**Comment by stevekrouse** [comment]
In order for me to help debug this issue for you, I'll need access to a running example of code that isn't working. Would it be possible for you to reproduce this issue in an open platform, such as jsfiddle?

For reference, my current guesses are:

1) The way that react re-renders the page is causing the Woof library to be loaded twice, and thus conflict with itself.

2) The way I'm detecting the `global=false` attribute is by first [finding the script tag through dubious means](https://github.com/stevekrouse/WoofJS/blob/master/src/woof.es6.js#L1780), which might not work given the way you're adding the Woof script tag.

One way to quickly solve your issue without any further debugging on my part would be to simply remove the code that runs Woof in global mode. Want to give that a try? Simply add the following script tag instead:

```html
<script src="https://cdn.rawgit.com/stevekrouse/WoofJS/c160971f0caef12f302cef71975b380091248aa3/dist/woof.js"></script>

**Comment by alabrazi** [comment]
> One way to quickly solve your issue without any further debugging on my part would be to simply remove the code that runs Woof in global mode. Want to give that a try? Simply add the following script tag instead:
> 
> <script src="https://cdn.rawgit.com/stevekrouse/WoofJS/c160971f0caef12f302cef71975b380091248aa3/dist/woof.js"></script>

Yes this solution worked, thanks!.  I have tried it before already, but I didn't want Woofjs methods to be globally seen in my application.

Based on your explanation, I think I will need mainly Reactjs to reproduce the problem. I can set an example on Jsfiddle or Codepen later.

**Comment by stevekrouse** [comment]
Hey @alabrazi - I came up with a better way to disable global mode than monkey patching the code. Just set `window.WOOF_GLOBAL_MODE = false` before the Woof library loads. 

I have it set up in a pull request now, so you can try it by using the new link

```
https://cdn.rawgit.com/stevekrouse/WoofJS/54fc6f6b9ab5033aaef7893710d98739f8bad35c/dist/woof.js
```

Let me know if that works for you and I'll get it merged into master, so you can get off of a monkey patch.

**Comment by alabrazi** [comment]
@stevekrouse thanks for the update. So far it is still not working. Or maybe I just don't know how to make it work. I added window.WOOF_GLOBAL_MODE = false to the main.html in the App and to componentDidMount before making new Woof instance. Then tried after new Woof instance ..... but it didn't work.

**Comment by stevekrouse** [comment]
The key is that `window.WOOF_GLOBAL_MODE = false` needs to happen before the WoofJS library is *loaded*, so you may have to put it before you import the Woof library. I'm not sure how your app is setup in order to make that happen...

You won't need it in our `componentDidMount`. It needs to happen before the library is loaded, not before making a new woof instance.
